openwisp_controller/config/tests/test_selenium.py

```diff
@@ -130,6 +132,49 @@ def test_create_new_device(self):
             'The Device “11:22:33:44:55:66” was added successfully.',
         )
 
+    def test_device_preview_overlay(self):
+        config = self._create_config(device=self._create_device(name='Test'))
+        config.device.full_clean()
+        config.device.save()
+        config.refresh_from_db()
+        self.login()
+        self.open(reverse('admin:config_device_changelist'))
+        self.web_driver.find_element(
+            by=By.CSS_SELECTOR, value='tbody tr:nth-child(1) th a'
+        ).click()
+        try:
+            WebDriverWait(self.web_driver, 2).until(
+                EC.presence_of_element_located((By.CSS_SELECTOR, '#content-main'))
+            )
+        except TimeoutException:
+            self.fail("Device detail page did not load in time")
+        actions = ActionChains(self.web_driver)
+        actions.key_down(Keys.ALT).send_keys('p').key_up(Keys.ALT).perform()
+        try:
+            WebDriverWait(self.web_driver, 2).until(
+                lambda driver: driver.find_element(
+                    By.CSS_SELECTOR, '.djnjc-overlay'
+                ).value_of_css_property('display')
+                != 'none'
+            )
+            overlay_visible = True
+        except TimeoutException:
+            overlay_visible = False
+        self.assertTrue(overlay_visible)
+        actions = ActionChains(self.web_driver)
+        actions.send_keys(Keys.ESCAPE).perform()
+        try:
+            WebDriverWait(self.web_driver, 2).until(
+                lambda driver: driver.find_element(
+                    By.CSS_SELECTOR, '.djnjc-overlay'
+                ).value_of_css_property('display')
+                == 'none'
```

Comment:
This assertion is ineffective and passes always, even if the `closePreview` JS function is broken on purpose.
Lesson: always break the tested code on purpose to verify the test catches the failure.